### PR TITLE
:sparkles: Add MachineDeployment example

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Release note**:

```release-note

```

### CODE REVIEW

File: examples/cluster/cluster.yaml [hunk 1]
```diff
@@ -4,6 +4,12 @@ kind: Cluster
 metadata:
   name: ${CLUSTER_NAME}
 spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.96.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
```
Comment (detiber) [inline examples/cluster/cluster.yaml]: services and serviceDomain can be omitted now.

File: examples/machinedeployment/machinedeployment.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
+  template:
+    metadata:
+      labels:
+        cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
+    spec:
+      version: v1.14.4
```
Comment (detiber) [inline examples/machinedeployment/machinedeployment.yaml]: Should we also template the k8s version?

Comment (vincepri) [reply examples/machinedeployment/machinedeployment.yaml]: Done!

File: examples/cluster/cluster.yaml (hunk 1 above)
Comment (vincepri) [reply examples/cluster/cluster.yaml]: Removed!

File: examples/machinedeployment/machinedeployment.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+kind: MachineDeployment
+metadata:
+  name: ${CLUSTER_NAME}-md-0
+  labels:
+    cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
```
Comment (rudoi) [inline examples/machinedeployment/machinedeployment.yaml]: x-k8s?

Comment (vincepri) [reply examples/machinedeployment/machinedeployment.yaml]: Argh, good catch

File: examples/generate.sh [hunk 4]
```diff
@@ -27,8 +27,9 @@ command -v "${ENVSUBST}" >/dev/null 2>&1 || echo -v "Cannot find ${ENVSUBST} in
 CLUSTERAWSADM=${CLUSTERAWSADM:-clusterawsadm}
 command -v "${CLUSTERAWSADM}" >/dev/null 2>&1 || echo -v "Cannot find ${CLUSTERAWSADM} in path, build it using 'make binaries' in this repository."
 
-# Cluster name.
+# Cluster.
 export CLUSTER_NAME="${CLUSTER_NAME:-test1}"
+export KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.14.4}"
```
Comment (detiber) [inline examples/generate.sh]: We should probably update the other existing templates for the k8s version env variable as well.

File: examples/machinedeployment/machinedeployment.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        kind: AWSMachineTemplate
+        name: ${CLUSTER_NAME}-md-0
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: AWSMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-md-0
+spec:
+  instanceType: ${CONTROL_PLANE_MACHINE_TYPE}
+  iamInstanceProfile: "control-plane.cluster-api-provider-aws.sigs.k8s.io"
```
Comment (detiber) [inline examples/machinedeployment/machinedeployment.yaml]: This should be `nodes.cluster-api-provider-aws.sigs.k8s.io` instead of `control-plane`

**Pull request merged**